cmd/syncComparator.go

```diff
@@ -65,13 +65,12 @@ type syncDestinationComparator struct {
 
 	comparisonHashType common.SyncHashType
 
-	preferSMBTime             bool
-	disableComparison         bool
-	deleteDestinationFileSync bool
+	preferSMBTime     bool
+	disableComparison bool
 }
 
-func newSyncDestinationComparator(i *objectIndexer, copyScheduler, cleaner objectProcessor, comparisonHashType common.SyncHashType, preferSMBTime, disableComparison bool, deleteDestinationFile bool) *syncDestinationComparator {
-	return &syncDestinationComparator{sourceIndex: i, copyTransferScheduler: copyScheduler, destinationCleaner: cleaner, preferSMBTime: preferSMBTime, disableComparison: disableComparison, comparisonHashType: comparisonHashType, deleteDestinationFileSync: deleteDestinationFile}
+func newSyncDestinationComparator(i *objectIndexer, copyScheduler, cleaner objectProcessor, comparisonHashType common.SyncHashType, preferSMBTime, disableComparison bool) *syncDestinationComparator {
```

Comment:
During sync we do delete file if its not present on the other side, are we revoking this feature now ?

Reply:
This is a confusion of two very similarly named flags that have significantly different functionality.

There are two flags vaguely named "delete destination". One of them cleans up files that are not present at the destination, the other deletes an object with uncommitted blocks before writing to the destination.

The former is used to basically mirror the source to the destination.

The latter is used to handle cases like writing to a blob with other tools, or writing to a blob prior written by the dfs endpoint.

Reply:
In this case, I've removed the latter, not the former. The former still happens in the destinationCleaner-- which is handled by the sync deleter.